invader_payment/services/invader_payment_service.py

```diff
@@ -34,14 +35,28 @@ def _invader_find_payable_from_transaction(self, transaction):
         """
         raise NotImplementedError()
 
+    def _invader_restrict_payment_mode_ids(self):
```

Comment:
we don't need it

Reply:
Indeed I should only have this in shopinvader_payment module.
I will refactor it

Reply:
but why?

Reply:
But we need to define which payment mode is allowed, right now you can paid with a payment mode that i snot configured on shopinvader backend

Reply:
the available payment_mode ids could depend on a lot of parameters. In some cases the available payment_mode will depend on the sale order itself. (mode if total € > ....).

Reply:
@sebastienbeau The validation should be at 2 levels:
Into the payment service implmentation we should check that the payment_mode_id received by each payment service is supporetd by the service.
When the payment_mode_id is applied to the payable, the payable could validate if this payment_mode_id can be applied. For the second point I would avoid to declare this check into invader.payable

Reply:
So the idea will to have (with stripe example)
- a check in "def confirm_payment" that check that the payment mode selected have stripe as provider.
- a check in model/sale.py of shopinvader_payment  that check that the payment_mode selected is compatible with the mode allowed on the backend? we add this in the "_invader_prepare_payment_transaction_data" ?

Reply:
Regarding making payment_mode_id requirer, we add this in every payment service (stripe, manual, sips), do we copy this or do we try to share it between all service?

Reply:
> So the idea will to have (with stripe example)
> 
>     * a check in "def confirm_payment" that check that the payment mode selected have stripe as provider.
> 
:+1: 

>     * a check in model/sale.py of shopinvader_payment  that check that the payment_mode selected is compatible with the mode allowed on the backend? we add this in the "_invader_prepare_payment_transaction_data" ?

:+1:

Reply:
> Regarding making payment_mode_id requirer, we add this in every payment service (stripe, manual, sips), do we copy this or do we try to share it between all service?

I prefer to keep the api without dependencies.... -> copy

Reply:
Ok I am going to do this I push it.
thanks for your usefull comment